gordon/metrics/log.py

```diff
@@ -0,0 +1,173 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Gordon ships with a default IMetricRelay implementation, which
+outputs metrics to the application logger using the standard
+library `logging` module.
+
+This implementation is used by default if no other IMetricRelay
+implementation is loaded and chosen.
+
+The default configuration that can be overriden is shown below.
+
+.. code-block:: ini
+
+    [metrics-logger]
+    # One of the log levels available from the stdlib logging module.
+    log_level = 'info'
+    # Time unit used by LogTimer. Can be one of 's', 'ms', 'ns'.
+    time_unit = 's'
+"""
+
+import collections
+import logging
+import time
+
+import zope.interface
+
+from gordon import interfaces
+
+# TODO: update plugin loader to load metrics logger as default
+
+
+@zope.interface.implementer(interfaces.ITimer)
+class LogTimer:
+    """Timer that outputs metrics to the logger.
+
+    Args:
+        metric (dict): Metric to output.
+        logger (.LoggerAdapter): Object used for outputting metrics.
+        time_unit (str): Unit for LogTimer output.
+            Options: s, ms, ns.
+    """
+    def __init__(self, metric, logger, time_unit):
+        self.metric = metric
+        self.logger = logger
+        self.time_multiplier = self._time_unit_to_multiplier(time_unit)
+
+    def _time_unit_to_multiplier(self, unit):
+        return {
+            's': 1,
+            'ms': 1E3,
+            'ns': 1E9
+        }[unit]
+
+    async def __aenter__(self):
+        await self.start()
+        return self
+
+    async def __aexit__(self, *args):
+        await self.stop()
+
+    async def start(self):
+        """Start the timer."""
+        self._start_time = time.perf_counter()
+
+    async def stop(self):
+        """Stop the timer and output the time delta to the logger."""
+        time_elapsed = (
+            time.perf_counter() - self._start_time) * self.time_multiplier
+        self.metric['value'] = time_elapsed
+        self.logger.log(self.metric)
+
+
+class LoggerAdapter:
+    """Configurable adapter that formats and outputs metrics.
+
+    Args:
+        level (str): Log level at which to output metric.
+    """
+    LOGFMT = '[{metric_name}] value: {value}'
+
+    def __init__(self, level):
+        self.level = level
+        self._logger = logging.getLogger('gordon.metrics-logger')
+
+    def log(self, metric):
+        """Format and output metric.
+
+        Args:
+            metric (dict): Complete metric.
+        """
+        message = self.LOGFMT.format(**metric)
+        if metric['context']:
+            message += ' context: {context}'.format(context=metric['context'])
+        getattr(self._logger, self.level)(message)
```

Comment:
At some point, you'll need to `lower()` the `self.level` in case it's configured in caps:

```pycon
>>> logger = logging.getLogger('')
>>> func = getattr(logger, 'INFO')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'RootLogger' object has no attribute 'INFO'
>>> func = getattr(logger, 'info')
>>>
```